addons/link_tracker/tools/html.py

```diff
@@ -0,0 +1,62 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import re
+from typing import Iterable
+
+import lxml
+
+MAX_LABEL_LENGTH = 40  # arbitrary
+
+
+def find_links_with_urls_and_labels(root_node, base_url, skip_regex=None, skip_prefix=None, skip_list=None):
+    """Return lxml link nodes and respective matching absolute urls and labels found in `root_node`.
+
+    :param lxml.etree._Element root_node: The root node to process
+    :param str base_url: base url to prefix relative hrefs
+    :param str skip_regex: URL pattern to skip
+    :param str skip_prefix: str prefix to skip
+    :param Iterable[str] skip_list: URLS to skip
+
+    :rtype: tuple[list[lxml.etree._Element], list[tuple[str, str]]]
```

Comment:
Could return an Ordereddict, keys = node, value = {'url': absolute_url, 'label': label} maybe ? Easier to manipulate than a tuple whose second item is also a tuple